x-pack/plugins/apm/public/components/shared/charts/latency_chart/index.tsx

```diff
@@ -99,7 +108,11 @@ export function LatencyChart({ height }: Props) {
           height={height}
           fetchStatus={latencyChartsStatus}
           id="latencyChart"
-          timeseries={latencyTimeseries}
+          customTheme={comparisonChartTheme}
+          timeseries={[
+            ...currentPeriod,
+            ...(comparisonEnabled && previousPeriod ? [previousPeriod] : []),
+          ]}
```

Comment:
I'm confused why these are spread. I guess currentPeriod is an array of timeseries? If so, should it be?

Reply:
> I guess currentPeriod is an array of timeseries? If so, should it be?

Yes, currentPeriod is an Array. I see your point of it not being an array but instead an object, since it will always return one option based on the latency aggregation type that is selected. But I don't think I'll have the time to refactor it. I'd rather create a new issue and tackle it on another PR, if that's ok for you.

Reply:
Now that we have more time, can we untangle this a bit?